### PR TITLE
[Java] Added a few more tests for Decimal to String cast

### PR DESCRIPTION
**Pull request by razajafri**
This PR adds a few more edge cases as a sanity test on the request of @sameerz

### CODE REVIEW

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 1]
```diff
@@ -3376,16 +3376,27 @@ void testFixedWidthCast() {
   void testCastBigDecimalToString() {
     BigDecimal[] bigValues = {new BigDecimal("923121331938210123.321"),
         new BigDecimal("9223372036854775808.191"),
+        new BigDecimal("-9.223"),
```
Comment (sameerz) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:3379]: Should we add a test case for a negative decimal value greater than 64 bits?

Comment (razajafri) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:3379]: The only thing that will accomplish is to verify that Java BigDecimals > 64 bit have precision > 38

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 2]
```diff
@@ -3376,16 +3376,27 @@ void testFixedWidthCast() {
...
-             "9328323982309091029831.002")) {
+             "-9.223",
+             "0.000",
+            "9328323982309091029831.002")) {
       assertColumnsAreEqual(expected, values);
     }
+
+    BigDecimal[] bigValues0 = {new BigDecimal("992983283728193827182918744829283742232")};
+    try {
+      ColumnVector cv = ColumnVector.fromDecimals(bigValues0);
```
Comment (revans2) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:3396]: Should we make sure that this didn't leak?

Comment (ttnghia) [reply java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:3396]: Obviously we should!

**Comment by razajafri** [comment]
rerun tests

**Review by revans2 — APPROVED** [review]

File: java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java [hunk 3]
```diff
@@ -3376,16 +3376,30 @@ void testFixedWidthCast() {
...
+      if (cv != null) {
+        cv.close();
+      }
```
Comment (ttnghia) [inline java/src/test/java/ai/rapids/cudf/ColumnVectorTest.java:3399]: Why not `try(....cv = ..) {}`? It is auto-closed, isn't it?

**Comment by razajafri** [comment]
@gpucibot merge

**Pull request merged**